src/services/class.service.ts

```diff
@@ -48,7 +51,7 @@ export class ClassService {
   async unregister(request: IClassRequest): Promise<string> {
     const { author, categoryType, className } = request;
     try {
-      if (categoryType === null) {
+      if (categoryType === undefined) {
```

Comment:
in register func, ur evaluating `!variable` i would use the same expression here